src/fifelse.c

```diff
@@ -358,12 +360,13 @@ SEXP fcaseR(SEXP na, SEXP rho, SEXP args) {
       }
     } break;
     case STRSXP: {
-      const SEXP *restrict pouts = STRING_PTR(outs);
-      const SEXP pna = nonna ? STRING_PTR(na)[0] : NA_STRING;
+      const SEXP *restrict pthens;
```

Comment:
Is `restrict` legal here? I'm honestly not sure. I'm 70% sure it's fine but would like a third pair of eyes. My concern is to do with the global string pool.

Reply:
I also _think_ it's fine, but OTOH, I can't tell whether `STRING_PTR()` is part of the API would be dinged eventually...